bokehjs/src/lib/models/graphs/layout_provider.ts

```diff
@@ -21,4 +22,75 @@ export abstract class LayoutProvider extends Model {
   abstract get_node_coordinates(graph_source: ColumnarDataSource): [Arrayable<number>, Arrayable<number>]
 
   abstract get_edge_coordinates(graph_source: ColumnarDataSource): [Arrayable<number>[], Arrayable<number>[]]
+
+  get node_coordinates(): NodeCoordinates {
+    return new NodeCoordinates({layout: this})
+  }
+
+  get edge_coordinates(): EdgeCoordinates {
+    return new EdgeCoordinates({layout: this})
+  }
+}
+
+export namespace GraphCoordinates {
+  export type Attrs = p.AttrsOf<Props>
+  export type Props = CoordinateTransform.Props & {
+    layout: p.Property<LayoutProvider>
+  }
+}
+
+export interface GraphCoordinates extends GraphCoordinates.Attrs {}
+
+export abstract class GraphCoordinates extends CoordinateTransform {
+  override properties: GraphCoordinates.Props
+
+  constructor(attrs?: Partial<GraphCoordinates.Attrs>){
```

Comment:
A single space is required before opening a block. There are multiple other instances to fix. I will add a lint rule for this.